### PR TITLE
feat: add `validationSchemaExportType` to config w/ resolve deps

### PR DESCRIPTION
**Pull request by Code-Hex**
fix: https://github.com/Code-Hex/graphql-codegen-typescript-validation-schema/pull/382#issuecomment-1579631051

I've added logic to graph the dependencies that GraphQL schemas have, and perform a [topological sort](https://en.wikipedia.org/wiki/Topological_sorting) based on that in order to resolve the dependencies. Normally, this sort is effective when there are no circular references, but I have made some modifications because users may generate code from schemas that have circular references.

### CODE REVIEW

**Comment by Code-Hex** [comment]
@elijaholmos Thank you!

**Comment by Code-Hex** [comment]
#164

**Comment by elijaholmos** [comment]
@Code-Hex this is awesome! thank you so much for this hard work that will surely introduce much expandability in the future!

File: example/zod/schemas.ts [hunk 1]
```diff
@@ -11,108 +11,84 @@ export const isDefinedNonNullAny = (v: any): v is definedNonNullAny => v !== und
...
     name: z.string().min(5),
     value: z.string().regex(/^foo/, "message")
-  })
-}
+});
 
-export function EventInputSchema(): z.ZodObject<Properties<EventInput>> {
-  return z.object<Properties<EventInput>>({
-    arguments: z.array(z.lazy(() => EventArgumentInputSchema())),
+export const EventInputSchema: z.ZodObject<Properties<EventInput>> = z.object({
```
Comment (tobiasdiez) [inline example/zod/schemas.ts:32]: is there a reason why this is no longer passing `Properties<EventInput>` to `z.object`? 

I now get errors of the form
```
Type 'ZodObject<{ deviceName: ZodEffects<ZodAny, any, any>; filePath: ZodEffects<ZodAny, any, any>; }, "strip", ZodTypeAny, { deviceName?: any; filePath?: any; }, { ...; }>' is not assignable to type 'ZodObject<Required<{ deviceName: ZodType<string, any, string>; filePath: ZodType<string, any, string>; }>, UnknownKeysParam, ZodTypeAny, { ...; }, { ...; }>'.
  Type '{ deviceName?: any; filePath?: any; }' is not assignable to type '{ deviceName: string; filePath: string; }'.
```
which are fixed if I provide the correct type to `object`.

**Pull request merged**